"""
Задание 2.

Создайте собственный!!! класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля
в качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.
"""
class ZeroError(Exception):
    def __init__(self, a):
        self.a = a

n_1 = int(input("Введите делимое: "))
n_2 = int(input("Введите делитель: "))

try:
    if n_2 == 0:
        raise ZeroError("Делить на 0 нельзя")
    otv = n_1 / n_2
except ZeroError as err:
    print(err)
else:
    print(f"Результат = {otv}")
